app/controllers/handoff/BasicCompanyDetailsController.scala

```diff
@@ -70,7 +69,7 @@ class BasicCompanyDetailsController @Inject()(val authConnector: PlayAuthConnect
       ctAuthorisedHandoff("HO1b", request) {
         registeredHandOff("HO1b", request) { _ =>
           handBackService.processCompanyNameReverseHandBack(request).map {
-            case Success(_) => Redirect(controllers.reg.routes.CompletionCapacityController.show)
+            case Success(payload) => Redirect(controllers.reg.routes.CompletionCapacityController.show).withLang(Lang((payload \ "language").as[String]))
```

Comment:
Just to remove some duplication, I think it might make sense to create a method within `handOffUtils` that returns the Lang from the given JSON. So something like:

```
val readLang: JsValue => Lang =
    _.as[Option[String]](Reads.nullable(__ \ "language")).fold(Lang(LangConstants.english))(Lang(_))
```

Then you could call that here `.withLang(handOffUtils.readLang(payload))` and in all other other reverse routing controllers.

Note, I also put the fallback to English just in case they don't send the language payload and that way at least the User can continue without an ISE being thrown.

Reply:
The only way the payload wouldn't contain the language is if the payload was missing or failed to decrypt. In that case it would throw an error page instead of trying to go on to the normal redirect. Is it wise to default a language to anything and not just maintain whatever the user has set on our side since we would have no way to tell actual updated preference?

I like the other idea though will make the changes

Reply:
If we have an ability to preserve the language choice from our side and use that as the default fallback then that would make more sense - I prefer and like that idea 😄 👍   

I do agree it would be unlikely to happen, but if it did _(say CH accidentally released something that turned off the sending of the language flag from their side)_ I think it does make sense to try and design in and build gracefulness where we have an opportunity to.

It would be much better for the User to be able to continue with their journey in one of the two languages _(as they can always flip the toggle again)_. It's much worse if we throw an ISE just because we didn't receive a language flag back from CH

Reply:
Awesome, updated for your comments and added the default to be whatever the Cookie is set too on our side :)